mysql/resource_grant.go

```diff
@@ -688,20 +697,42 @@ func ImportGrant(ctx context.Context, d *schema.ResourceData, meta interface{})
 	return results, nil
 }
 
+func produceGrantId(database string, username string, host string) string {
+	if strings.Compare(database, "*") != 0 && !strings.HasSuffix(database, "`") {
```

Comment:
Maybe `database != "*"`? That would be a bit more readable for the first part.

Reply:
sure thing, I just copied that over from the, now defunct, `formatDatabaseName()` function :)